R/PqConnection.R

```diff
@@ -135,3 +135,38 @@ postgresWaitForNotify <- function(conn, timeout = 1) {
 postgresIsTransacting <- function(conn) {
   connection_is_transacting(conn@ptr)
 }
+
+
+#' Imports a large object from file
+#'
+#' Returns an object idenfier (Oid) for the imported large object
+#'
+#' @export
+#' @param conn a [PqConnection-class] object, produced by
+#'   [DBI::dbConnect()]
+#' @param filepath a path to the large object to import
+#' @param oid the oid to write to. Defaults to 0 which assigns an unused oid
+#' @return the identifier of the large object, an integer
+#' @examples
+#' con       <- postgresDefault()
+#' path_to_file <- 'my_image.png'
+#' dbWithTransaction(con, { 
+#'  oid <- postgresImportLargeObject(con, test_file_path)
+#' })
+postgresImportLargeObject <- function(conn, filepath = NULL, oid = 0) {
+
+  if (!postgresIsTransacting(conn)) {
+    stopc("Cannot import a large object outside of a transaction")
+  }
+
+  if (is.null(filepath)) stopc("'filepath' cannot be NULL")
+  if (oid <  0)     stopc("'oid' cannot be negative")
+  if (is.null(oid) | is.na(oid)) stopc("'oid' cannot be NULL/NA")
```

Comment:
```suggestion
  if (is.null(oid) || is.na(oid)) stopc("'oid' cannot be NULL/NA")
```

Or perhaps this could be two checks?